File: Development/Web UI/sql-dm-widgets/src/main/java/com/idera/sqldm_10_3/data/LicenseDetails.java
package com.idera.sqldm_10_3.data;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.idera.common.rest.DataContractDateDeserializer;
import com.idera.common.rest.DataContractUtcDateSerializer;

import java.util.Date;

@JsonIgnoreProperties(ignoreUnknown = true)
public class LicenseDetails {

	@JsonProperty("Expiration")
	@JsonDeserialize(using = DataContractDateDeserializer.class)
	@JsonSerialize(using = DataContractUtcDateSerializer.class)
	private Date expiration;
	public Date getExpiration() {
		return expiration;
	}

	@JsonProperty("IsPermanent")
	private boolean isPermanent;
	public boolean isPermanent() {
		return isPermanent;
	}

	@JsonProperty("IsTrial")
	private boolean isTrial;
	public boolean isTrial() {
		return isTrial;
	}
	
	@JsonProperty("LicensedServers")
	private int licensedServers;
	public int getLicensedServers() {
		return licensedServers;
	}
	
	@JsonProperty("CheckedKeyCount")
	private int CheckedKeyCount;
	public int getCheckedKeyCount() {
		return CheckedKeyCount;
	}
	
	@JsonProperty("MonitoredServers")
	private int monitoredServers;
	public int getMonitoredServers() {
		return monitoredServers;
	}
	
	@JsonProperty("Repository")
	private String repository;
	public String getRepository() {
		return repository;
	}

	//Enum Status { OK, Expired, CountExceeded, NoValidKeys }
	@JsonProperty("Status")
	private String status;
	public String getStatus() {
		return status;
	}

	public long getDaysToExpire() {
		long diff=expiration.getTime()-System.currentTimeMillis();
		return  diff / (24 * 60 * 60 * 1000);
	}
}

File: Development/Idera/rameshwar/SQLdm/Service/DataContracts/v1/License/LicenseDetails.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;
using Idera.SQLdm.Common;

namespace Idera.SQLdm.Service.DataContracts.v1.License
{
    [DataContract]
    public class LicenseDetails
    {
        private DateTime _expiration;

        [DataMember]
        public int CheckedKeyCount { get; set; }
        
        [DataMember]
        public DateTime Expiration
        {
            get 
            {
                return (_expiration==DateTime.MinValue || _expiration == DateTime.MaxValue)?new DateTime(2099,12,31):_expiration;
            }
            set
            {
                _expiration = value;
            }
        }

        [DataMember]
        public string Status { get; set; }
        [DataMember]
        public string Repository { get; set; }
        [DataMember]
        public int MonitoredServers { get; set; }
        [DataMember]
        public int LicensedServers { get; set; }
        [DataMember]
        public bool IsUnlimited { get; set; }
        [DataMember]
        public bool IsTrial { get; set; }
        [DataMember]
        public bool IsPermanent { get; set; }
    }    
}
